### PR TITLE
[enhance](mtmv)Provide a method to determine whether incremental updates of multiple tables are allowed

### PR DESCRIPTION
**Pull request by zddr**
Currently, only range partitioning is supported, and it must be a dynamic partition or auto partition.

At the same time, it is required that the partitioning rules of different tables be consistent

Think about the list partition carefully before making any decisions

### CODE REVIEW

**Comment by zddr** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/mtmv/MTMVPartitionCheckUtil.java [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+import java.util.Objects;
+
+public class MTMVPartitionCheckUtil {
+    /**
+     * Check if the partitioning method of the table meets the requirements for multi table partitioning updates
+     *
+     * @param tableIf base table of materialized view
+     * @return Inspection results and reasons
+     */
+    public static Pair<Boolean, String> checkIfAllowMultiTablePartitionRefresh(MTMVRelatedTableIf tableIf) {
```
Comment (seawinde) [inline fe/fe-core/src/main/java/org/apache/doris/mtmv/MTMVPartitionCheckUtil.java:40]: tableIf is relatedTable may be better

File: fe/fe-core/src/main/java/org/apache/doris/mtmv/MTMVPartitionCheckUtil.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        } else if (isAutoPartition(originalTable)) {
+            return compareAutoPartition(originalTable, relatedTable);
+        } else {
+            throw new AnalysisException("only support dynamic/auto partition");
+        }
+    }
+
+    @VisibleForTesting
+    public static Pair<Boolean, String> compareDynamicPartition(OlapTable originalTable,
+            OlapTable relatedTable) throws AnalysisException {
```
Comment (seawinde) [inline fe/fe-core/src/main/java/org/apache/doris/mtmv/MTMVPartitionCheckUtil.java:76]: `OlapTable relatedTable` is `List<OlapTable>relatedTable` may be better

**Comment by zddr** [comment]
run buildall

**Review by seawinde — APPROVED** [review]

**Review by morrySnow — APPROVED** [review]

**Comment by zddr** [comment]
run performance

**Pull request merged**